### PR TITLE
chore(samples): Code samples for Retail Tutorials. Search service samples (part 1)

### PR DESCRIPTION
**Pull request by sborisenkox**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-retail/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> ☕️

### CODE REVIEW

File: samples/interactive-tutorials/src/main/java/search/SearchSimpleQuery.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        SearchRequest.newBuilder()
+            .setPlacement(defaultSearchPlacementName)
+            .setQuery(queryPhrase)
+            .setVisitorId(visitorId)
+            .setPageSize(pageSize)
+            .build();
+    System.out.println("Search request: " + searchRequest);
+
+    try (SearchServiceClient client = SearchServiceClient.create()) {
+      SearchResponse searchResponse = client.search(searchRequest).getPage().getResponse();
```
Comment (Neenu1995) [inline samples/interactive-tutorials/src/main/java/search/SearchSimpleQuery.java:60]: For clarity's sake, should the method name be different?
`client.search` vs ` public static SearchResponse search()`

**Review by chenlei1216 — APPROVED** [review]

File: samples/interactive-tutorials/src/main/java/search/SearchSimpleQuery.java (hunk 1 above)
Comment (sborisenkox) [reply samples/interactive-tutorials/src/main/java/search/SearchSimpleQuery.java:60]: Yes, make sense.
Changed, also in a second PR part.

File: samples/interactive-tutorials/src/main/java/search/SearchWithBoostSpec.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            .setPlacement(defaultSearchPlacementName)
+            .setQuery(searchQuery)
+            .setVisitorId(visitorId)
+            .setBoostSpec(boostSpec)
+            .setPageSize(pageSize)
+            .build();
+    System.out.println("Search request: " + searchRequest);
+
+    try (SearchServiceClient client = SearchServiceClient.create()) {
+      SearchResponse searchResponse = client.search(searchRequest).getPage().getResponse();
```
[the remark this answers is not in the document]
Comment (tetiana-karasova) [reply samples/interactive-tutorials/src/main/java/search/SearchWithBoostSpec.java:71]: fixed

File: samples/interactive-tutorials/src/main/java/search/SearchWithFacetSpec.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+            .setPlacement(defaultSearchPlacementName)
+            .setQuery(searchQuery)
+            .setVisitorId(visitorId)
+            .addFacetSpecs(facetSpec)
+            .setPageSize(pageSize)
+            .build();
+    System.out.println("Search request: " + searchRequest);
+
+    try (SearchServiceClient client = SearchServiceClient.create()) {
+      SearchResponse searchResponse = client.search(searchRequest).getPage().getResponse();
```
[the remark this answers is not in the document]
Comment (tetiana-karasova) [reply samples/interactive-tutorials/src/main/java/search/SearchWithFacetSpec.java:62]: fixed

**Review by Neenu1995 — APPROVED** [review]

File: samples/interactive-tutorials/README.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+Default Credentials.
+
+1. Set your user credentials to authenticate your requests to the Retail API
+
+    ```bash
+    gcloud auth application-default login
+    ```
+
+1. Type `Y` and press **Enter**. Click the link in Terminal. A browser window should appear asking
+   you to log in using your Gmail account.
```
Comment (kweinmeister) [inline samples/interactive-tutorials/README.md:41]: Suggest changing this to "log in using your Google account"

Comment (sborisenkox) [reply samples/interactive-tutorials/README.md]: Done.

**Review by kweinmeister — APPROVED** [review]

**Review by kweinmeister — APPROVED** [review]

**Review by Neenu1995 — APPROVED** [review]

**Review by kweinmeister — APPROVED** [review]

**Pull request merged**